src/plugins/data/common/interpreter/expression_types/kibana_context.ts

```diff
@@ -17,17 +17,15 @@
  * under the License.
  */
 
-import { Query } from 'src/legacy/core_plugins/data/public';
-import { TimeRange } from 'ui/timefilter/time_history';
 import { Filter } from '@kbn/es-query';
 
 const name = 'kibana_context';
 
 export interface KibanaContext {
   type: typeof name;
-  query?: Query;
+  query?: any; // any -> import { Query } from 'src/legacy/core_plugins/data/public';
   filters?: Filter[];
-  timeRange?: TimeRange;
+  timeRange?: any; // any -> import { TimeRange } from 'ui/timefilter/time_history';
```

Comment:
Are these comments here as a reminder to change things back later? Did you run into issues with the imports?

Reply:
We cannot import from NP plugin legacy platform files. `src/legacy` will be deleted and `ui/...` is `src/legacy/ui/...`. I will take a look, if those are just TypeScript interfaces, I can probably already move them to NP, otherwise this is something to be changed later when those things are moved over.

Reply:
we don't need to (or rather shouldn't) move all the functions and types together with interpreter.

for example kibana_context funciton has nothing to do with interpreter .... 
- visualization functions should live with their appropriave vis_type
- data functions should live with appropriate data services
- app specific functions should live with their apps.

Reply:
@lukeelmers kibana_datatable and esaggs functions will be part of same plugin (`data`) but actually other service (`search`) ... i think we should put those functions together with each service they belong to.

Reply:
Using `TimeRange` and `Query` types now.

Reply:
@ppisljar Yep I think that makes complete sense. The only caveat is that for TS interfaces, those will obviously need to be re-exported at the top level with everything else (this is mainly just needed for interpreter types that get registered... the interfaces for those will need to be consumable in other plugins)

But otherwise there is no reason why everything needs to stay living together vs separated into the correct location like we've done with the vis type functions.